gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java

```diff
@@ -0,0 +1,526 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.examples.dns;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.dns.ChangeRequest;
+import com.google.gcloud.dns.Dns;
+import com.google.gcloud.dns.DnsOptions;
+import com.google.gcloud.dns.DnsRecord;
+import com.google.gcloud.dns.ProjectInfo;
+import com.google.gcloud.dns.Zone;
+import com.google.gcloud.dns.ZoneInfo;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An example of using Google Cloud DNS.
+ *
+ * <p>This example creates, deletes, gets, and lists zones, and creates and deletes DNS records of
+ * type A.
+ *
+ * <p>Steps needed for running the example:
+ * <ol>
+ * <li>login using gcloud SDK - {@code gcloud auth login}.</li>
+ * <li>compile using maven - {@code mvn compile}</li>
+ * <li>run using maven - {@code mvn exec:java
+ * -Dexec.mainClass="com.google.gcloud.examples.dns.DnsExample"
+ * -Dexec.args="[<project_id>]
+ * create <zone_name> <dns_name> <description> |
+ * get <zone_name> |
+ * delete <zone_name> |
+ * list [<zone_name> [changes [descending | ascending] | records]] |
+ * add-record <zone_name> <record_name> <ip> <ttl> |
+ * delete-record <zone_name> <record_name> <ip> [<ttl>] |
+ * quota}</li>
+ * </ol>
+ *
+ * <p>The first parameter is an optional {@code project_id} (logged-in project will be used if not
```

Comment:
Nit: instead of "logged-in", maybe you could say "project specified in the Google Cloud SDK configuration (see `gcloud config list`) will be ..."